chainer/functions/evaluation/binary_accuracy.py

```diff
@@ -0,0 +1,44 @@
+import numpy
+
+from chainer import cuda
+from chainer import function
+from chainer.utils import type_check
+
+
+class BinaryAccuracy(function.Function):
+    def check_type_forward(self, in_types):
+        type_check.expect(in_types.size() == 2)
+        x_type, t_type = in_types
+
+        type_check.expect(
+            x_type.dtype == numpy.float32,
+            t_type.dtype == numpy.int32,
+            t_type.shape[0] == x_type.shape[0],
```

Comment:
Further, I think we should impose more tighter restriction on the shapes of `x` and `t`. `t_type.shape == x_type.shape`, that is, to impose that `x` and `t` must have same shape. How do you think of it?